goty-server/src/main/java/com/aleinin/goty/TitleService.java

```diff
@@ -0,0 +1,57 @@
+package com.aleinin.goty;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.ImmutableList;
+import lombok.Builder;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.context.ApplicationListener;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.lang.NonNull;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+class TitleService implements ApplicationListener<ContextRefreshedEvent> {
+
+    private ImmutableList<Game> games = ImmutableList.of();
+
+    private final ObjectMapper objectMapper;
+
+    public List<String> search(String word, int count) {
+        return games.stream()
+                .filter(game -> game.getNames()
+                        .stream()
+                        .anyMatch(name -> StringUtils.containsIgnoreCase(name, word)))
+                .limit(count)
+                .map(Game::getName)
+                .collect(toImmutableList());
+    }
+
+    @Override
+    public void onApplicationEvent(@NonNull ContextRefreshedEvent event) {
+        try {
+            games = ImmutableList.copyOf(objectMapper.readValue(new File("game_titles.json"), Game[].class));
+        } catch (IOException e) {
+            log.error(e.getMessage(), e);
+        }
+    }
+
+    @Value
+    @Builder
+    private static class Game {
+
+        int id;
+        String name;
+        List<String> names;
```

Comment:
in what context would a game ave a name and a list of names?

Reply:
The IGDB API additionally returns "alternative titles" for a game. So for instance "TES5" and "TESV" are alternative titles for Skyrim. What I'm doing is including both the "primary" and "alternative" titles in the "names" field and searching on it. When there's a match, return the primary title only. So if a search was done for "TES5" the response would return "The Elder Scrolls V: Skyrim". 

I am actually working on some changes right now to make this all a little more clear and to improve the searching functionality.

Reply:
Yeah I'd recommend calling it "aliases" or "alternative names" or something. Your logic makes sense I just couldn't infer that from what was there

Reply:
"aliases" is good. 👍 

Names are hard lol

Reply:
@aleinin I will fix it in my next pr. I'm already pretty far along there.